assets/src/components/kubernetes/common/DeleteResource.tsx

```diff
@@ -60,10 +61,34 @@ enum DeletionPropagation {
   DeletePropagationOrphan = 'Orphan',
 }
 
+function toServerError(data: FetchResult): ServerError {
+  const defaultError: ServerError = {
+    statusCode: 500,
+    result: 'Could not delete resource',
+  } as ServerError
+
+  if (!data.errors) {
+    return defaultError
+  }
+
+  if (!((data.errors as unknown) instanceof ApolloError)) {
+    return defaultError
+  }
+
+  const apolloError: ApolloError = data.errors as unknown as ApolloError
+  const networkError: ServerError = apolloError?.networkError as ServerError
+
+  return {
+    statusCode: networkError?.statusCode,
+    result: networkError?.result ?? networkError?.message,
+  } as ServerError
+}
+
 function DeleteResourceModal({ open, setOpen, resource, refetch }): ReactNode {
   const theme = useTheme()
   const [deleting, setDeleting] = useState(false)
   const [deleteNow, setDeleteNow] = useState(false)
+  const [serverError, setServerError] = useState<ServerError>()
```

Comment:
why can't you use the error field of the mutation hook for this?

Reply:
That's because the actual error message is part of the response and the error provided directly by apollo does not have the response message. I think that's due to how apollo rest link translates rest responses into GQL. The only place where an error response is available is inside the response `data` returned by the mutation under `errors` field. It took me quite some time to figure out where the error response is stored and how to read it.

Reply:
Oh very weird

Reply:
Unfortunately the `data` returned by the `useMutation` hook is different than the one returned by the mutation promise returned when calling the mutation.

Normally I would get it like that but this `data` is undefined when there is an error when calling the mutation, and `error` does not contain the REST request response.
```
const [deleteResource, { error, data }] = useMutation(...)
```

The `data` that I ended up using is returned by the mutation function promise
```
deleteResource().then((data) => setServerError(toServerError(data)))
```

Reply:
whoa ok that's also weird

Reply:
That is probably due to how apollo rest link plugin handles response transformation and error handling. I haven't seen any better way that would allow us to handle it in a generic way.